### PR TITLE
Update Cybervillains root CA to use keysize=2048 and SHA256. Also add a Common Name

### PR DESCRIPTION
**Pull request by ehyche**
This fixes https://jira.groupondev.com/browse/IOSCON-45777

By default, the CA certificate and server certificates that Odo was generating on the 9090 port were:

* RSA 1024 bits
* Signed with SHA1

In order to meet the ATS requirements specified here:

https://developer.apple.com/library/archive/documentation/General/Reference/InfoPlistKeyReference/Articles/CocoaKeys.html#//apple_ref/doc/uid/TP40009251-SW57

Then the certificates needed to be:

* RSA >= 2048 bits
* Signed with SHA256 or greater

So since the code that handled these parameters was in the browsermob-proxy, then we pulled the following files from the 2.0.0-beta9 tag of browsermob-proxy into our own com.groupon.browsermob-proxy:

* KeyStoreManager.java
* CertificateCreator.java
* ExtendedKeyUsageConstants.java

And made the modifications there. In addition, we had to let the code upgrade the following files:

proxyui/src/main/resources/cybervillainsCA.jks
proxyui/src/main/resources/cybervillainsCA.cer

These are the files which contain the CA certificate and private key for the Certificate Authority.

Also, we discovered a bug in iOS where if a CA certificate does not have a Common Name in its X.509 Distinguished Name, then it will not show up in Settings/About/Certificate Trust Settings and thus the user cannot manually trust the CA certificate. So we added a Common Name to the Cybervillains CA Distinguished Name.

### CODE REVIEW

File: browsermob-proxy/src/main/java/com/groupon/odo/bmp/CertificateCreator.java [hunk 1]
```diff
@@ -0,0 +1,413 @@
+package com.groupon.odo.bmp;
```
Comment (djrenfro) [inline browsermob-proxy/src/main/java/com/groupon/odo/bmp/CertificateCreator.java]: Can you add a licensing comment to the top of these files similar to https://github.com/groupon/odo/blob/master/browsermob-proxy/src/main/java/com/groupon/odo/bmp/BrowserMobProxyHandler.java#L1

**Comment by djrenfro** [comment]
Can you put comments in the code where you actually tweaked the original files?

File: browsermob-proxy/src/main/java/com/groupon/odo/bmp/CertificateCreator.java (hunk 1 above)
Comment (ehyche) [reply browsermob-proxy/src/main/java/com/groupon/odo/bmp/CertificateCreator.java]: Yep, will do.

**Comment by ehyche** [comment]
@djrenfro : will add comments on the places where code is changed.

**Comment by ehyche** [comment]
@djrenfro : Updated the PR per your review comments.

**Pull request merged**